engine/common/rpc/convert/convert.go

```diff
@@ -1212,6 +1185,43 @@ func MessageToTrieUpdate(m *entities.TrieUpdate) (*ledger.TrieUpdate, error) {
 	}, nil
 }
 
+// TrieUpdateToMessage converts a TrieUpdate to a protobuf message
+func TrieUpdateToMessage(t *ledger.TrieUpdate) (*entities.TrieUpdate, error) {
+	if t == nil {
+		return nil, nil
+	}
```

Comment:
```suggestion
```

A best practice is that when a function takes a pointer, it only handle the case where the pointer isn't `nil`, and the caller has to handle the nil case. 

That's why I'm suggesting to move the handling to the caller. 

```
	if data.TrieUpdate == nil {
		return nil, nil
	}
	trieUpdate, err := TrieUpdateToMessage(data.TrieUpdate)


```

Reply:
ah got it!